bots/pr/src/main/java/org/openjdk/skara/bots/pr/CommandHandler.java

```diff
@@ -34,8 +33,8 @@
 interface CommandHandler {
     String description();
 
-    default void handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInstance, Path scratchPath, CommandInvocation command, List<Comment> allComments, PrintWriter reply)
-    {
+    default void handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInstance, Path scratchPath, CommandInvocation command,
+                        List<Comment> allComments, PrintWriter reply, List<String> labelsToAdd, List<String> labelsToRemove) {
     }
```

Comment:
I keep coming back to this in my reviewing. It would be better if we did not have to update all those other places where handle() is implemented. 

I might be thinking wrong here, but I believe that you should be able to keep the original `handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInstance, Path scratchPath, CommandInvocation command, List<Comment> allComments, PrintWriter reply)` method, and instead add the new `handle` (with `List<String> labelsToAdd, List<String> labelsToRemove` as a default method, with the implementatation
```
handle(bot, pr, censusInstance, scratchPath, command, allComments, reply);
```

This way, the commands that need the new label handling mechanism can implement the new handle override with the extended argument list. All other commands needs no change.

Once again, I don't feel 100% confident I'm thinking correctly here, but it seems to me like it would work, and be a better solution.

Reply:
You are correct Magnus, that would be nicer indeed.

Reply:
The drawback of doing that is that JEPCommand would still need to implement the old handle method, which doesn't make sense for it to do.

Reply:
> This way, the commands that need the new label handling mechanism can implement the new handle override with the extended argument list. All other commands needs no change.

If we obey this convention which adds a new "bridge" `handle` method. Next time, we want to add another parameter and add a new `handle` method, too. The interface `CommandHandler` will have sevaral such "bridge" `handle` methods in the future which are actually unnecessary. @erikj79, @magicus and I may feel good with it because we added or reviewed it. But in the future, the new SKARA developers may be confused by it. In other words, the maintainment of the project may become a little more complex.

>  but I'm also ok leaving it as it is.

I would like to keep the code as it is now.

Reply:
No, each and every implementation of `CommandHandler` would only implement a single `handle()` method.

At least, if I'm thinking correctly about this. :-)

When processing WorkItems, in `processCommand`, we call the "long" version, with the two extra arguments. This will have a default implementation in the `CommandHandler` interface, which calls the "short" version, dropping the two last arguments. The "short" version will have a default implementation, which does nothing. (This is exactly the case of the code right now, without this PR.) 

So, my suggestion is that we leave the old "short" method untouched, add a new "long" method which by default delegates to the "short" method, and call the new "long" method in `processCommand`.

Then all old implementations can keep their "short" implementations, and everything will work as before.

JEPCommand will need to override the new "long" version instead (and can just ignore the "short" version), and everything will work perfect for it too.

And if we in the future need to add further arguments, we can do the same procedure again -- add a new long-long method, which deletages to the long method, etc.

I believe this was the exact use case that the `default` method implementation in interfaces was intended for.

Reply:
And I believe this solution is so much better that I'd strongly argue that we should not keep the code as it looks right now. There is no need nor good reason to push these additional argument to command implementations.

Reply:
> @erikj79, @magicus and I may feel good with it because we added or reviewed it. But in the future, the new SKARA developers may be confused by it. In other words, the maintainment of the project may become a little more complex.

@magicus previously, I thought it is not good to add the "bridge" "long" methods because it may confused other developers.

> There is no need nor good reason to push these additional argument to command implementations.

But now, I think that the additional parameter `labelsToAdd` and `labelsToRemove` which have not used in much cammands confused the developers, too and are more terrible. So I agree with you and updated the code just now.